test/functional/sqlsrv/sqlsrv_ae_insert_sqltype_int_conv.phpt

```diff
@@ -0,0 +1,124 @@
+--TEST--
+Test for inserting encrypted data of int types
+--DESCRIPTION--
+Test implicit conversions between different integer types
+With Always Encrypted, implicit conversion works if:
+1. From input SQLSRV_SQLTYPE_BIT to a bit column
+2. From input SQLSRV_SQLTYPE_BIT to a tinyint column
+3. From input SQLSRV_SQLTYPE_BIT to a smallint column
+4. From input SQLSRV_SQLTYPE_BIT to an int column
+5. From input SQLSRV_SQLTYPE_BIT to a bigint column
+6. From input SQLSRV_SQLTYPE_TINYINT to a tinyint column
+7. From input SQLSRV_SQLTYPE_TINYINT to a smallint column
+8. From input SQLSRV_SQLTYPE_TINYINT to an int column
+9. From input SQLSRV_SQLTYPE_TINYINT to a bigint column
+10. From input SQLSRV_SQLTYPE_SMALLINT to a smallint column
+11. From input SQLSRV_SQLTYPE_SMALLINT to an int column
+12. From input SQLSRV_SQLTYPE_SMALLINT to a bigint column
+13. From input SQLSRV_SQLTYPE_INT to an int column
+14. From input SQLSRV_SQLTYPE_INT to a bigint column
+15. From input SQLSRV_SQLTYPE_BIGINT to a bigint column
+Without AlwaysEncrypted, inplicit conversion between different integer types works
+--SKIPIF--
+<?php require('skipif_versions_old.inc'); ?>
+--FILE--
+<?php
+require_once('MsCommon.inc');
+
+$dataTypes = array("bit", "tinyint", "smallint", "int", "bigint");
+$sqlTypes = array("SQLSRV_SQLTYPE_BIT", "SQLSRV_SQLTYPE_TINYINT", "SQLSRV_SQLTYPE_SMALLINT", "SQLSRV_SQLTYPE_INT", "SQLSRV_SQLTYPE_BIGINT");
+$inputValues = array(1, 0);
+
```

Comment:
The same input values for all types?

Reply:
this test is for testing different integer conversions. if the input was 20, it wouldn't fit into a bit, and the conversion would fail not because of unsupported conversion, but because of errors like truncation, and the test is not targeting. For a test with different inputs, see sqlsrv_ae_insert_sqltype_numeric.phpt

Reply:
please put your justification in a comment so that we will remember in the future